apisix/init.lua

```diff
@@ -456,6 +456,8 @@ function _M.http_access_phase()
                 api_ctx.matched_route = route
                 core.table.clear(api_ctx.plugins)
                 api_ctx.plugins = plugin.filter(api_ctx, route, api_ctx.plugins)
+                -- rerun rewrite phase for newly added plugins in consumer
+                plugin.rerun_plugins_of_consumer(api_ctx.plugins, api_ctx)
```

Comment:
Currently, plugins in APISIX can only be run once, but after introducing this method, some plugins (in rewrite phase) can be run twice.

This inconsistence will confuse the users. I think either we only allow running each plugin for just one time or we need to redesign the plugin system so that plugins can be run multiple times.

Reply:
In fact, no plug-in will run twice in this case. The reason for this misunderstanding is that

1. APISIX runs the rewrite phase of plugins on route first;
2. APISIX handles the logic for consumer authentication based on the auth plugins on the route;
3. then APISIX merges the plugins on consumer and route; 
4. this causes the rewrite phase to be missed for some non-auth plugins configured on the consumer.

Reply:
Thanks for the explanation.